### PR TITLE
[DUOS-2112][risk=low] Add support for searching mail sent by date range.

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-2112

[DUOS-2112](https://broadworkbench.atlassian.net/browse/DUOS-2112) builds on work completed as part of https://github.com/DataBiosphere/consent/pull/1862 and makes the following enhancements:

- Endpoint and class updates for searching mail by date range added.
- Documentation and tests added to describe and validate the feature.
- Limit and Offset are now validated to be greater than -1 for both searching by EmailType and Date Range.

### Notes
This PR is a branch of https://github.com/DataBiosphere/consent/pull/1864

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

[DUOS-2112]: https://broadworkbench.atlassian.net/browse/DUOS-2112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/resources/assets/paths/mailByRange.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      required: false
+      schema:
+        type: string
+    - name: limit
+      in: query
+      description: Number of results to return.  Defaults to 20 if unset. May not be negative
+      required: false
+      schema:
+        type: integer
+    - name: offset
```
Comment (connorlbark) [inline src/main/resources/assets/paths/mailByRange.yaml:26]: feedback from before, but i am just curious on thoughts: any interest in an optional parameter to filter by type as well? i feel usually you would be debugging / trying to find specific emails of a known type when searching for date as well.

**Review by connorlbark — APPROVED** [review]
this is much cleaner, and i like the new tests a lot more. i will go ahead and approve because it works great and code is good. optional feedback is optional, up to what you think the use case of this is

File: src/main/resources/assets/paths/mailByRange.yaml (hunk 1 above)
Comment (rushtong) [reply src/main/resources/assets/paths/mailByRange.yaml:26]: Filtering on type is a different use case that we don't actually have yet. I can foresee the need for it, but I would prefer that we build that feature when we actually need it.

**Review by quazi-h — APPROVED** [review]
Changes make sense to me, tests passed when running locally. I was unable to find the /mail endpoints when running consent locally, I made sure that my user had an Admin role when checking /user/me. Is there a way to view/test those endpoints on swagger locally?

**Comment by rushtong** [comment]
> I was unable to find the /mail endpoints when running consent locally, I made sure that my user had an Admin role when checking /user/me. Is there a way to view/test those endpoints on swagger locally?

The endpoint is defined in both `Admin` and `Mail` headings as seen here:
Screen Shot 2023-01-10 at 2 24 07 PM

**Pull request merged**